cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_nodegroup_test.go

```diff
@@ -651,95 +677,102 @@ func TestNodeGroupDeleteNodes(t *testing.T) {
 	// sorting and the expected semantics in test() will fail.
 
 	t.Run("MachineSet", func(t *testing.T) {
-		test(t, newMachineSetTestObjs(t.Name(), 0, 10, 10, map[string]string{
+		test(t, mustCreateMachineSetTestConfig(testNamespace, 10, 10, map[string]string{
 			nodeGroupMinSizeAnnotationKey: "1",
 			nodeGroupMaxSizeAnnotationKey: "10",
 		}))
 	})
 
 	t.Run("MachineDeployment", func(t *testing.T) {
-		test(t, newMachineDeploymentTestObjs(t.Name(), 0, 10, 10, map[string]string{
+		test(t, mustCreateMachineDeploymentTestConfig(testNamespace, 10, 10, map[string]string{
 			nodeGroupMinSizeAnnotationKey: "1",
 			nodeGroupMaxSizeAnnotationKey: "10",
 		}))
 	})
 }
 
 func TestNodeGroupMachineSetDeleteNodesWithMismatchedNodes(t *testing.T) {
-	nreplicas := 1
+	test := func(t *testing.T, expected int, testConfigs []*testConfig) {
+		t.Helper()
+		testConfig0, testConfig1 := testConfigs[0], testConfigs[1]
+		controller, stop := mustCreateTestController(t, testConfigs...)
+		defer stop()
 
-	machineSet0 := makeMachineSet(0, nreplicas, map[string]string{
-		nodeGroupMinSizeAnnotationKey: "1",
-		nodeGroupMaxSizeAnnotationKey: "3",
-	})
+		nodegroups, err := controller.nodeGroups()
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+		if l := len(nodegroups); l != expected {
```

Comment:
I am always opposed to one letter variables. Such variables should be used only for indexes. Otherwise, I would go with `nodegroupsLength` or just `ngLen` instead here.